phishingpage.py

```diff
@@ -0,0 +1,138 @@
+
+# set of dictionaries to store URLs and names
+CISCO = {"name1": "url1", "name2": "url2"}
+TEMPLATE_TO_URL = {"cisco": CISCO}
+
+PHISHING_PAGES_DIR = "phishing-scenarios/"
+
+
+class UrlNotAvailable(Exception):
+    """ Exception class to raise in case of a invalid URL. """
+    def __init__(self):
+        Exception.__init__(self, "The given URL is not available!")
+
+
+class TemplateNotAvailable(Exception):
+    """ Exception class to raise in case of non-existence template. """
+    def __init__(self):
+        Exception.__init__(self, "The given template is not available!")
+
+
+class AurgumentIsNotAString(Exception):
+    """ Exception class to raise in case of non-string input. """
+    def __init__(self):
+        Exception.__init__(self, "The given argument is not a string!")
+
+
+def grab_online(template):
+    """
+    Make a new folder with the name provided in template and fetch all the
+    files for the given template from the internet.
+
+    Args:
+        template (str): The name of requested template.
+
+    Returns:
+        None
+
+    Raises:
+        TemplateNotAvailble: If the given template is not in the database.
+        AurgumentIsNotAString: if the given template is not a string.
+    """
+    # check to see if template is a string
+    if not type(template) is str:
+        raise AurgumentIsNotAString()
+
+    # remove the directory and get the name of the template
+    chosen_template = template[template.find("/")+1:]
+
+    # check if user's choice exists in dictionary
+    if chosen_template in TEMPLATE_TO_URL:
+
+        # make a new folder
+        os.makedirs(chosen_template)
+
+        # get the keys of second dictionary
+        for name in TEMPLATE_TO_URL[chosen_template]:
+
+            # get the URL
+            url = TEMPLATE_TO_URL[chosen_template][name]
+
+            # check if URL's exist
+            if url_checker(url):
+                # download the files
+                urllib.urlretrieve(url, template + "/" + name)
+
+    else:
+
+        raise TemplateNotAvailable()
+
+
+def exists(dir_path):
+    """
+    Check if the template directory in the given path exists.
+
+    Args:
+        dir_path (str): path of a directory.
+
+    Returns:
+        True: if directory exists.
+        False: if directory does not exist.
+
+    Raises:
+        AurgumentIsNotAString: if the given path is not a string.
+    """
+    # check to see if given path is a string
+    if not type(dir_path) is str:
+        raise AurgumentIsNotAString()
+
+    # check the path and return accordingly
+    if os.path.isdir(dir_path):
+        return True
+    else:
+        return False
+
+
+def get(template_name):
+    """
+    Return the directory of the the given template.
+
+    Args:
+        template_name (str): The name of the chosen template.
+
+    Returns:
+        (str): The full path of the chosen template.
+
+    Raises:
+        AurgumentIsNotAString: if the given template name is not a string.
+    """
```

Comment:
There should be a check here that the template actually exists using the `exists` method. I know that `wifiphisher.py` will also perform this check but that doesn't bother us. We need to think of this module as a stand-alone script.